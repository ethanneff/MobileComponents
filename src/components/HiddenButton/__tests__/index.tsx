import React from "react";
import renderer from "react-test-renderer";
import Component from "..";
import { setE2EModeChange } from "../../../commons/Store/Debug";
import { getTestProvider } from "../../../commons/Utils/TestMocks/get-test-provider";

describe("HiddenButton Component", () => {
  it("renders correctly with defaults", () => {
    const { TestProvider } = getTestProvider();
    const component = renderer.create(
      <TestProvider>
        <Component onPress={jest.fn()} />
      </TestProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders correctly with  props", () => {
    const { TestProvider, store } = getTestProvider();
    store.dispatch(setE2EModeChange(true));
    const component = renderer.create(
      <TestProvider>
        <Component onPress={jest.fn()} testID="123" />
      </TestProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
