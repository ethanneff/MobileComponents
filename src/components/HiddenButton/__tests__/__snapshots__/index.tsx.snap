// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenButton Component renders correctly with  props 1`] = `
<TouchableOpacity
  onPress={[MockFunction]}
  style={
    Object {
      "height": 2,
      "width": 2,
    }
  }
  testID="123"
/>
`;

exports[`HiddenButton Component renders correctly with defaults 1`] = `null`;
